src/__experimental__/components/input/input-presentation.style.js

```diff
@@ -0,0 +1,68 @@
+import styled, { css } from 'styled-components';
+import PropTypes from 'prop-types';
+import baseTheme from '../../../style/themes/base';
+import OptionsHelper from '../../../utils/helpers/options-helper';
+import sizes from './input-sizes.style';
+import inputClassicStyling from './input-presentation-classic.style';
+
+const InputPresentationStyle = styled.div`
```

Comment:
Just thought... What do you think about split the code on two parts 'hardcode' and 'variable based code'. For now we have something like mixing variables with hardcoded values, what do you think about this solution?
```
const InputPresentationStyle = styled.div`
  align-items: center;
  background: transparent;
  box-sizing: border-box;
  cursor: text;
  display: flex;
  flex-wrap: wrap;
  flex: 1;
  margin: 0px;
  width: 100%;

  ${({
    theme, size, disabled, readOnly, hasFocus
  }) => css`
    border: 1px solid ${theme.colors.border};
    min-height: ${sizes[size].height};
    padding-left: ${sizes[size].padding};
    padding-right: ${sizes[size].padding};

    ${disabled
      && css`
        background: ${theme.disabled.input};
        border-color: ${theme.disabled.border};
        cursor: not-allowed;
      `}

    ${readOnly
      && css`
        background: transparent !important;
        border-color: transparent !important;
      `}

    ${hasFocus
      && css`
        && {
          outline: 3px solid ${theme.colors.focus};
        }
      `}
  `}

  ${stylingForValidation('info')}
  ${stylingForValidation('warning')}
  ${stylingForValidation('error')}

  ${inputClassicStyling}
`;
```

Reply:
I like this, but it might get more complicated when we start trying to follow it for components that have nested selectors or pseudo selectors? It may be a convention that is too restrictive or difficult to follow in those cases?

Reply:
Yup, you're right. It's can be a problem